.github/workflows/release.yaml

```diff
@@ -0,0 +1,32 @@
+name: release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+permissions:
+  actions: write
+  contents: write
+  pull-requests: write
+  statuses: write
+
+jobs:
+  Release:
+    runs-on: ubuntu-latest
+    # don't allow for publishing release unless everything else passes
+    # TODO: not sure if this will work
+    # needs: [Build, Prettier, Unit-Tests, E2E-Tests, Dockerfile-Lint, ESLint, Type-Check]
```

Comment:
There are two ways to approach this.

1. Require that we only tag a commit that is known to be good (i.e., it's already landed on `main` and passed CI).  This is generally OK, since a release is a manual process (i.e., you tag).  Also, what if these fail?  You've already created the tag, so you need to do another patch release anyway.
2. You can separate your jobs out into separate files, and call them via `uses`.  We do this in Telescope, for example: https://github.com/Seneca-CDOT/telescope/blob/master/.github/workflows/release.yml#L14.  This is nice for re-use, but sucks for maintenance, since you can't update the jobs in a PR, and have to land changes first before you can test things.

I'd be tempted to do 1.

Reply:
I agree, 1 seems like the better option. Since releasing is a manual process, we should have the freedom, if push comes to shove, to release with CI not passing (when that prettier job randomly started failing). I don't think its within the scope of this job to enforce all CI must pass before we can publish a release. That decision should be on the actual person creating the release.